### PR TITLE
fix : horizontal field: label and values are not vertically aligned 

### LINKED ISSUE #30470: [Bug]: horizontal field: label and values are not vertically aligned
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (32) x64 13th Gen Intel(R) Core(TM) i9-13900K
    Memory: 68.75 GB / 95.75 GB
  Browsers:
    Edge: Chromium (121.0.2277.98)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/affectionate-beaver-8ddjk8?file=%2Fsrc%2FApp.js

## Actual Behavior
Label and value are not vertically aligned: The label is higher up than the value in the input.
This is visible on your webpage: https://react.fluentui.dev/?path=/docs/components-field--default#horizontal-orientation.
Just enter a value in the input, then scroll the content up until the label disappears: the value will still be partly visible.
The provided reproduction is not a real one - it is only so that this issue is not auto closed.
By the way: Repreductions are hard when you dont link to an existing template. And codesandbox seems to loose free usage.

## Expected Behavior
Label and values should vertically align.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

Correcting the issue seems to be as simple as adding `align-items: center` to the Field grid.

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by vicheanath**
## Previous Behavior
Screenshot 2024-02-09 at 11 48 38 AM

## New Behavior
Screenshot 2024-02-09 at 11 48 25 AM

## Related Issue(s)

- Fixes #30470

### CODE REVIEW

File: packages/react-components/react-field/src/components/Field/useFieldStyles.styles.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ const useLabelStyles = makeStyles({
 
   horizontal: {
     marginRight: tokens.spacingHorizontalM,
+    lineHeight: tokens.lineHeightBase500,
```
Comment (behowell) [inline packages/react-components/react-field/src/components/Field/useFieldStyles.styles.ts:61]: We can't change the line height, as that will affect the second+ line of the label text if it wraps.

**Review by behowell — CHANGES_REQUESTED** [review]
Hi @vicheanath, thank you for the PR. The current visuals match the design spec, so we can't take this change without a design update. Putting a block on this PR for now. I'll follow up on #30470.

**Comment by vicheanath** [comment]
thanks for letting mw know

File: packages/react-components/react-field/src/components/Field/useFieldStyles.styles.ts (hunk 1 above)
Comment (vicheanath) [reply packages/react-components/react-field/src/components/Field/useFieldStyles.styles.ts:61]: let me try with padding

**Comment by behowell** [comment]
This will be fixed by #30732. (Note that PR is currently waiting on a final design, but will be merged once we have the padding values from design.)

**Pull request closed without merging**